src/traccuracy/_tracking_graph.py

```diff
@@ -316,20 +316,27 @@ def get_connected_components(self) -> list[TrackingGraph]:
         return [self.get_subgraph(g) for g in nx.weakly_connected_components(graph)]
 
     def get_subgraph(self, nodes: Iterable[Hashable]) -> TrackingGraph:
-        """Returns a new TrackingGraph with the subgraph defined by the list of nodes
+        """Returns a new TrackingGraph with the subgraph defined by the list of nodes.
 
         Args:
-            nodes (list): List of node ids to use in constructing the subgraph
-        """
+            nodes (list): A non-empty list of node ids to use in constructing the subgraph
 
+        Raises:
+            ValueError if nodes is empty
+        """
         new_graph = self.graph.subgraph(nodes).copy()
+        logger.debug(f"Subgraph has {new_graph.number_of_nodes()} nodes")
+        if new_graph.number_of_nodes() == 0:
```

Comment:
The only issue is you can techncially pass nodes that aren't in the graph to subgraph, and it will take the intersection of the passed list of nodes and those in the graph. So a len(nodes) != 0 does not necessarily mean the subgraph will have nodes. This may not matter if we allow empty TrackingGraphs as discussed below

Reply:
Ah I see - ok let's wait on that discussion and then resolve this.

Reply:
This check has been removed, since we now allow empty TrackingGraphs